eclipse/tern.eclipse.ide.server.nodejs.core/src/tern/eclipse/ide/server/nodejs/core/NodejsDebuggersManager.java

```diff
@@ -0,0 +1,47 @@
+/**
+ *  Copyright (c) 2015 Genuitec LLC.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Piotr Tomiak <piotr@genuitec.com> - initial API and implementation
+ */
+package tern.eclipse.ide.server.nodejs.core;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import tern.eclipse.ide.server.nodejs.internal.core.debugging.webclipse.WebclipseDebugger;
+
+public class NodejsDebuggersManager {
+
+	private static Map<String, INodejsDebugger> debuggers = new LinkedHashMap<String, INodejsDebugger>();
+
+	static {
+		// Add supported debuggers
+		addDebugger(new WebclipseDebugger());
```

Comment:
> My big problem is that Webclipse will be available in the combo although users could not have this installed debugger.

Ok, so my idea is the opposite. I think that it's good for a user to see what options he has, otherwise he will not understand why he is not able to debug, or worse, he will not know which debugger to get. Note, I've update commit to have a validation logic for the field.

![tern-debug-not-installed](https://cloud.githubusercontent.com/assets/4244523/10395977/37d0dd92-6ea0-11e5-993d-dbc70241bb63.png)

Also, if there is interest from other debugger providers and they feel they need more flexibility, they can expand the framework. At this point it's hard to say what others would need. I usually refactor the code when it's needed, otherwise I can put a lot of unneeded effort.